source/data-modeling/field-types.txt

```diff
@@ -0,0 +1,558 @@
+.. _mongoid-field-types:
+
+===========
+Field Types
+===========
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: fields, data types, type conversion, code example
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn about the field types that you can use in
+{+odm+} to define the schema for your MongoDB documents. 
+
+MongoDB uses :manual:`BSON types </reference/bson-types/>` to represent the type of
+data stored in document fields. To use that data in a {+odm+} application,
+{+odm+} must convert the BSON types to {+language+} types at runtime. For example,
+when retrieving a document from the database, {+odm+} translates a BSON field
+with the type ``double`` to use the {+language+} ``Float`` type. It then translates that
+field back to a BSON ``double`` when saving the document to the database.
+
+.. note::
+
+   Changing the field definition in a model class does not automatically alter
+   the data stored in the database. To change the data type of a field in the
+   database, you must re-save the data after the change.
+
+Field Types
+-----------
+
+You can define the field type for fields in a model class by using the ``field``
+and ``type`` macros. The following example defines the fields of a ``Person``
+class:
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-define-fields
+   :end-before: # end-define-fields
+
+The following list describes the field types that you can use in {+odm+}:
+
+- ``Array``
+- ``Bson::Binary``
+- ``BigDecimal``
+- ``Mongoid::Boolean`` or ``"Boolean"``
+- ``Date``
+- ``DateTime``
+- ``Float``
+- ``Hash``
+- ``Integer``
+- ``Object``
+- ``Bson::ObjectId``
+- ``Range``
+- ``Regexp``
+- ``Set``
+- ``String``
+- ``Mongoid::StringifiedSymbol``
+- ``Time``
+- ``ActiveSupport::TimeWithZone``
+
+.. note::
+
+   {+odm+} does not support the ``BSON::Int64`` or ``BSON::Int32`` types.
+   Defining a field as one of these types will save to the database correctly,
+   but querying the field will return the fields as an ``Integer``. 
```

Comment:
I think the fact that is saves correctly is more due to MongoDB, not because of partial support from the driver. From my understanding, it doesn't support them at all as fields. I reworded this section to hopefully be a bit more clear